File: Practica 2/BACKEND/Typescript/src/models/Task.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";


@Entity({ name: "Task" }) 
export class Task {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  id_user!: number;

  @Column({ length: 150 })
  title!: string;

  @Column({ length: 300, nullable: true })
  description!: string;

  @Column({ default: false })
  status!: boolean;

  @Column({ 
    type: 'timestamp',
    default: () => 'CURRENT_TIMESTAMP'
  })
  created_at!: Date;
}

File: Practica 2/BACKEND/AspNet/AspNet/Models/Task.cs
using System;
using System.Collections.Generic;

namespace AspNet.Models;

public partial class Task
{
    public int Id { get; set; }

    public int IdUser { get; set; }

    public string Title { get; set; } = null!;

    public string Description { get; set; } = null;

    public bool? Status { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual User? IdUserNavigation { get; set; }
}
